internal/cmd/beta/security-group-rule/create/create_test.go

```diff
@@ -0,0 +1,342 @@
+			}),
+			isValid: false,
+		},
+		{
+			description: "protocol is icmp and parameters are missing",
+			flagValues: fixtureFlagValues(func(flagValues map[string]string) {
+				delete(flagValues, icmpParameterCodeFlag)
+				delete(flagValues, icmpParameterTypeFlag)
+			}),
+			isValid: false,
+		},
+		{
+			description: "protocol is icmp and port range values are provided",
+			flagValues:  fixtureFlagValues(),
+			isValid:     false,
+		},
+		{
+			description: "protocol is not icmp and port range values are provided",
+			flagValues: fixtureFlagValues(func(flagValues map[string]string) {
+				flagValues[protocolNameFlag] = "not-icmp"
+				delete(flagValues, icmpParameterCodeFlag)
+				delete(flagValues, icmpParameterTypeFlag)
+			}),
+			isValid: true,
+			expectedModel: fixtureInputModel(func(model *inputModel) {
+				model.IcmpParameterCode = nil
+				model.IcmpParameterType = nil
+				model.ProtocolName = utils.Ptr("not-icmp")
+			}),
+		},
+		{
+			description: "protocol is not icmp and icmp parameters are provided",
+			flagValues: fixtureFlagValues(func(flagValues map[string]string) {
+				flagValues[protocolNameFlag] = "not-icmp"
+			}),
+			isValid: false,
+		},
+		{
+			description: "no values",
+			flagValues:  map[string]string{},
+			isValid:     false,
+		},
+		{
+			description: "project id missing",
+			flagValues: fixtureFlagValues(func(flagValues map[string]string) {
+				delete(flagValues, projectIdFlag)
+			}),
+			isValid: false,
+		},
+		{
+			description: "project id invalid 1",
+			flagValues: fixtureFlagValues(func(flagValues map[string]string) {
+				flagValues[projectIdFlag] = ""
+			}),
+			isValid: false,
+		},
+		{
+			description: "project id invalid 2",
+			flagValues: fixtureFlagValues(func(flagValues map[string]string) {
+				flagValues[projectIdFlag] = "invalid-uuid"
+			}),
+			isValid: false,
+		},
+		{
+			description: "security group id missing",
+			flagValues: fixtureFlagValues(func(flagValues map[string]string) {
+				delete(flagValues, securityGroupIdFlag)
+			}),
+			isValid: false,
+		},
+		{
+			description: "security group id invalid 1",
+			flagValues: fixtureFlagValues(func(flagValues map[string]string) {
+				flagValues[securityGroupIdFlag] = ""
+			}),
+			isValid: false,
+		},
+		{
+			description: "security group id invalid 2",
+			flagValues: fixtureFlagValues(func(flagValues map[string]string) {
+				flagValues[securityGroupIdFlag] = "invalid-uuid"
+			}),
+			isValid: false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.description, func(t *testing.T) {
+			p := print.NewPrinter()
+			cmd := NewCmd(p)
+			err := globalflags.Configure(cmd.Flags())
+			if err != nil {
+				t.Fatalf("configure global flags: %v", err)
+			}
+
+			for flag, value := range tt.flagValues {
+				err := cmd.Flags().Set(flag, value)
+				if err != nil {
+					if !tt.isValid {
+						return
+					}
+					t.Fatalf("setting flag --%s=%s: %v", flag, value, err)
+				}
+			}
+
```

Comment:
In order to test also the MutuallyExlusive part you have to add:

```suggestion

			err = cmd.ValidateFlagGroups()
			if err != nil {
				if !tt.isValid {
					return
				}
				t.Fatalf("error validating flag groups: %v", err)
			}
```

Afterwards some of your tests will fail because both protocol name and protocol number are defined (e.g. base).